### PR TITLE
Added Login and SignUp Forms

### PR DESCRIPTION
**Pull request by Lalit3716**
Added Login and Signup forms with validations

### CODE REVIEW

File: components/Forms/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+				control => control.isValid,
+			);
+			return newState;
+		default:
+			return state;
+	}
+};
+
+const SignUp = () => {
+	const [form, dispatch] = useReducer(formReducer, {
```
Comment (supercoww) [inline components/Forms/SignUp.js:36]: A lot of code related to the state management and validation is repetitive such as the reducer and updating of touched state and deciding when to show an error. Can we abstract this into a useForm which can be used for any form? This hook would take in the initial value of each control, a validator function and provide the current value, error, isTouched and such details for each form control.
Alternative: Use a prebuilt solution such as https://react-hook-form.com/. If you choose this route, this is probably how you would need to integrate it with MUI components https://react-hook-form.com/get-started#IntegratingwithUIlibraries.

File: components/Forms/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+				isValid = value.trim().length > 0;
+				break;
+		}
+
+		dispatch({ type: 'FORM_CHANGE', control: controlName, value, isValid });
+	};
+
+	return (
+		<Container component="main" maxWidth="xs">
+			<CssBaseline />
```
Comment (supercoww) [inline components/Forms/SignUp.js:74]: No need to add CssBaseline on each page. It is already included globally in `_document.js`

**Review by supercoww — CHANGES_REQUESTED** [review]

File: components/Forms/SignUp.js (hunk 1 above)
Comment (Lalit3716) [reply components/Forms/SignUp.js:36]: Agreed, I myself didn't like how I did validation I will refactor all of this into a custom useForm hook.

File: components/Forms/SignUp.js (hunk 2 above)
Comment (Lalit3716) [reply components/Forms/SignUp.js:74]: Oops sorry I didn't noticed CssBaseline in ```_document.js```

**Comment by Lalit3716** [comment]
I did all the changes mentioned.

File: components/Forms/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+		event.preventDefault();
+		console.log('submit');
+	};
+
+	const onChangeHandler = (event, controlName) => {
+		const value = event.target.value;
+		let isValid = false;
+		switch (controlName) {
+			case 'email':
+				isValid = /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value);
```
[the remark this answers is not in the document]
Comment (Lalit3716) [reply components/Forms/SignUp.js]: Good to know!

**Pull request closed without merging**